src/test/ui/suggestions/unnamable-types.rs

```diff
@@ -0,0 +1,33 @@
+// Test that we do not suggest to add type annotations for unnamable types.
+
+#![crate_type="lib"]
+#![feature(generators)]
+
+const A = 5;
+//~^ ERROR: missing type for `const` item
+//~| HELP: provide a type for the item
+
+static B: _ = "abc";
+//~^ ERROR: the type placeholder `_` is not allowed within types on item signatures
+//~| NOTE: not allowed in type signatures
+//~| HELP: replace with the correct type
+
+
+const C: _ = || 42;
+//~^ ERROR: the type placeholder `_` is not allowed within types on item signatures
+//~| NOTE: not allowed in type signatures
+//~| NOTE: however, the inferred type
+
+struct S<T> { t: T }
+const D = S { t: || -> i32 { 42 } };
+//~^ ERROR: missing type for `const` item
+//~| NOTE: however, the inferred type
+
+fn foo() -> i32 { 42 }
+const E = S { t: foo };
+//~^ ERROR: missing type for `const` item
+//~| NOTE: however, the inferred type
```

Comment:
AFAIK, shouldn't this suggest the correct type instead: `S<fn() -> i32>`?

Reply:
Yeah, that could be a reasonable suggestion. @FabianWolff how do you feel about this?

Reply:
I have implemented it now; could you have another look to see whether it is what you had in mind?

I've also tried to do it for non-capturing closures, but for some reason, querying their signature always yielded something like
```rust
extern "rust-call" fn(()) -> i32
```
for a closure that doesn't take any arguments and returns an `i32`, which is not useful and wouldn't work as a suggestion.

Reply:
I mean, you could just change the would just changing the ABI to `Rust` not work? Also, can you be sure there is a test case for both capturing and non-capturing closures? And if you don't name the non-capturing closure here, maybe add a `FIXME`

Reply:
I have added two `FIXME`s for now, one in the code and one in the test case.